### DISCUSSION TITLE
Set path for Wix Toolset in windows runner

### DISCUSSION THREAD

**Discussion by hibearpanda** [community/community#27149] [Actions]
<p>I am trying to automate building a Windows MSI Installer for an electron application using Wix.</p>
<p>I can see from looking <a href="https://github.com/actions/virtual-environments/blob/e470bcde79524156356e18ca998cc2b40fbf3dc4/images/win/Windows2019-Readme.md" rel="noopener nofollow ugc">here</a> that the Wix Toolsets are included in the <code>windows-latest</code> runner.</p>
<p>However, it seems like <code>electron-wix-msi</code> requires having <code>candle.exe</code> and <code>light.exe</code> available in the path. Without the <code>candle.exe</code> and <code>light.exe</code> available in the path, my build step will fail with the following message:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/9/e9b652e8dbe0c9072eea66b98421c64cdf4b35e0.png" data-download-href="/uploads/short-url/xlvSpRlu3yQMEsisT5ChZoiQLRe.png?dl=1" title="Screenshot 2021-01-11 at 9.19.49 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/11482515/181102072-9f8c4b3b-e7e6-4925-97b3-3533cd9f3ee9.png" alt="Screenshot 2021-01-11 at 9.19.49 AM" data-base62-sha1="xlvSpRlu3yQMEsisT5ChZoiQLRe" width="690" height="191" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/9/e9b652e8dbe0c9072eea66b98421c64cdf4b35e0_2_690x191.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/9/e9b652e8dbe0c9072eea66b98421c64cdf4b35e0_2_1035x286.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/e/9/e9b652e8dbe0c9072eea66b98421c64cdf4b35e0_2_1380x382.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/9/e9b652e8dbe0c9072eea66b98421c64cdf4b35e0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-01-11 at 9.19.49 AM</span><span class="informations">1704×472 88.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>My research also lead to this <a href="https://github.com/felixrieseberg/electron-wix-msi/issues/1#issuecomment-674598795" rel="noopener nofollow ugc">post</a>, but the answer there uses <code>add-path</code> which has since been deprecated.</p>
<p>I have 2 questions:</p>
<ol>
<li>
<p>What is path to the wix toolset binaries? How can I access this in the <code>run</code> step? (<code>$wix</code>, <code>${env.wix}</code>?)</p>
</li>
<li>
<p>How can I add-path to the wix toolset so that it will be available at a later run step?</p>
</li>
</ol>
<p>Appreciate any help I can get!</p>

**Comment by otter-computer** [comment]
<p>Hi <a class="mention" href="/hibearpanda">@hibearpanda</a>, welcome to the GitHub Support Community!</p>
<p>The <code>::add-path</code> command has been replaced with an environment file, so you’d add something to your path using this syntax: <code>echo "/your/path/here" &gt;&gt; $GITHUB_PATH</code></p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#adding-a-system-path" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#adding-a-system-path" target="_blank" rel="noopener">Workflow commands for GitHub Actions - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#adding-a-system-path" target="_blank" rel="noopener">//docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>That said, I believe the wix toolset should already be installed by default and available in the path at <code>env.wix</code>, so this is rather strange. Would you be able to share a link to your workflow, or copy paste it in a public gist or something so I can take a look at exactly what it does?</p>

**Comment by hibearpanda** [comment, marked as answer]
<p>Hi <a class="mention" href="/thomasshaped">@thomasshaped</a>,</p>
<p>Thanks for replying. I was able to figure out how to add path but for some reason could not add the wix binaries to the path without hardcoding the path, instead of using env.wix. I have included a gist of list of my latest workflow <a href="https://gist.github.com/hibearpanda/d5b5b58bd319132ffd7af606ffa333fc" rel="noopener nofollow ugc">here</a>, but in summary:</p>
<p>Worked</p>
<pre><code class="lang-auto">      - name: Set path for candle and light
        run: echo "C:\Program Files (x86)\WiX Toolset v3.11\bin" &gt;&gt; $GITHUB_PATH
        shell: bash
</code></pre>
<p>Did not work:</p>
<pre><code class="lang-auto">      - name: Set path for candle and light
        env:
          WIX_PATH: ${{ env.wix }}
        run: echo "$env:WIX_PATH" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
</code></pre>
<pre><code class="lang-auto">      - name: Set path for candle and light
        env:
          WIX_PATH: ${{ env.wix }}
        run: echo "$env:WIX_PATH\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
</code></pre>

**Reply by erkstruwe** [reply]
It actually works using the `WIX` environment variable, too:

```
- name: Add WiX toolkit to PATH
  shell: bash
  run: echo "${WIX}bin" >> $GITHUB_PATH
```

**Comment by otter-computer** [comment]
<p>Oh nice, good to see you got it working! I also did some testing yesterday with some of my team members and ended up getting it to work via this method (though I don’t think this is a nice way to do it, it feels rather hacky):</p>
<pre><code class="lang-yaml">      - run: |
          $env:Path += ";C:\Program Files (x86)\WiX Toolset v3.11\bin"
          light.exe
</code></pre>
<p>It’s still very strange that they’re not available by default nor via using the preset env. Perhaps this is something to do with the specific env reference syntax differences between different shells used as I see you’ve managed to get it to work specifically when using bash.</p>
<p>I think I’m going to ask up the chain to see if they have any advice on this, but for now I’m at least glad you’ve managed to find solution!</p>

**Comment by hibearpanda** [comment]
<p>Yea, it is a bit strange that it was not available in the path. Certainly stranger that it’s not (seemingly) available with <code>env.wix</code>. I would have preferred to construct that path with <code>env.wix</code> so it isn’t hard coded but alas I have to move on.</p>
<p>Appreciate your help and for asking around!</p>
